### ISSUE TITLE
Type 'Noise' does not provide any constructors in C#

### ISSUE THREAD

**Issue by Drnagol369** [godotengine/godot#96396]
### Tested versions

Reproducible in Godot 4.3 stable dotnet and 4.2.2 stable dotnet, when using C#

### System information

Windows 11

### Issue description

From my limited experimentation, I assume that this is because the inbuilt Noise class is abstract. Whilst this isn't an issue with GDScript, it prevents C# scripts inheriting from this class, as the language - as far as I know - requires classes to have a constructor, and should provide one by default - although that appears not to be the case here, because its not an actual C# class, but a wrapper of some sort. IDK, I'm not really familiar with the inner workings of Godot.

![Screenshot 2024-08-31 152758](https://github.com/user-attachments/assets/cae18e3e-ba2e-4f3d-9131-0b76ec4784ae)
![Screenshot 2024-08-31 154742](https://github.com/user-attachments/assets/2c3007fe-6a88-40a3-9ad4-6ea729032b01)

### Steps to reproduce

Create a C# class that inherits from Noise, and try to build the project.

### Minimal reproduction project (MRP)

N/A

**Comment by Zylann** [comment]
This is actually a broader issue: the `Noise` cannot be extended by scripts or gdextensions. It is abstract, and none of its functions are virtual.

**Comment by Tsunjigiri** [comment]
> This is actually a broader issue: the `Noise` cannot be extended by scripts or gdextensions. It is abstract, and none of its functions are virtual. Maybe you can inherit from it in GDScript, but to little benefit due to the absence of implementable API. I have custom noises too in my project and I can't make them derive from Noise so they can't be used anywhere Noise is used, and vice versa.

Yeah, encountered this same when creating custom noise in GDExtension. 
Tried extending in GDScript right now, and it does cause an error as well.
`Class 'CustomNoise' cannot be constructed as it is based on abstract native class 'Noise'`
To be fair I didn't add any methods, but the context considered I doubt that would make any difference anyways.

Really defeats the purpose of an abstract interface to begin with if it cannot be implemented, but then again, this seems like a bug or an incomplete implementation, not the intention.
From [Noise description](https://docs.godotengine.org/en/stable/classes/class_noise.html#description) for example:
> This class defines the interface for noise generation libraries to inherit from.

From a cursory search around, there seems to have been some changes and discussions related to this stuff where some of these "abstract" classes have been made instantiable, but not all (e.g. CanvasItem). Some links I looked at:

- https://github.com/godotengine/godot/issues/71834
- https://github.com/godotengine/godot/issues/55549
- https://github.com/godotengine/godot/issues/35484
- https://github.com/godotengine/godot/pull/58972
- https://github.com/godotengine/godot/pull/67510

Importantly, [here's a comment from that PR](https://github.com/godotengine/godot/pull/58972#issuecomment-1063839813) with a list of the updated classes that presumably work since, and Noise isn't there.

It would really seem that it should be extendable though, as it wouldn't make sense to extend a custom noise class from a class with existing implementations (unless you wanted to use them). The description I quoted above explicitly implies this as well, unlike several other abstract class documentations.

As such, it does indeed seem like a missing implementation for now.

**Comment by raulsntos** [comment]
Changing the label from https://github.com/godotengine/godot/labels/topic%3Adotnet to https://github.com/godotengine/godot/labels/topic%3Acore, since the C# bindings generator is working as expected as per the ClassDB registration. If we want to make `Noise` instantiable, the ClassDB registration needs to be changed to allow that and that's a Core concern.

https://github.com/godotengine/godot/blob/ca871ccc9cd8e5c2ce829ec5c0fae1ac32c6269f/modules/noise/register_types.cpp#L52
